omnibus/package-scripts/ddot-rpm/preinst

```diff
@@ -0,0 +1,124 @@
+    fi
+}
+
+stop_agent
+deregister_agent
+
+# RPM unpacks the new files before running prerm of the old package
+# triggering manually the prerm python script of the old package
+if [ -f "$INSTALL_DIR/embedded/bin/python" ]; then
+    if [ -f "${INSTALL_DIR}/python-scripts/pre.py" ]; then
+        PRE_PYTHON_FILE="${INSTALL_DIR}/python-scripts/pre.py"
+    elif [ -f "${INSTALL_DIR}/python-scripts/prerm.py" ]; then
+        PRE_PYTHON_FILE="${INSTALL_DIR}/python-scripts/prerm.py"
+    fi
+
+    if [ -n "$PRE_PYTHON_FILE" ]; then
+        # -B prevents writing a cache of the bytecode since this is only run once
+        ${INSTALL_DIR}/embedded/bin/python -B "${PRE_PYTHON_FILE}" "${INSTALL_DIR}" || true
+    else
+        echo "[ WARNING ]\tPRE_PYTHON_FILE is not set"
+    fi
+fi
+
+# Starting with 6.10, integrations are also uninstalled on package removal
+
+# Since 6.18.0, a file containing all integrations files which have been installed by
+# the package is available. We use it to remove only the datadog-related check files which
+# have *NOT* been installed by the package (eg: installed using the `integration` command).
+
+if [ -f "$INSTALL_DIR/embedded/.installed_by_pkg.txt" ]; then
+    echo "Removing integrations installed with the 'agent integration' command"
+
+    # List all files in the embedded dir of the datadog-agent install dir
+    PREV_DIR=$(pwd)
+    cd $INSTALL_DIR
+    find . -depth -path './embedded/lib/python*/site-packages/datadog_*' >$INSTALL_DIR/embedded/.all-integrations.txt
+
+    # List all files in the embedded dir of the datadog-agent install dir
+    # which were not installed by the package and rm them.
+    grep -Fxv -f $INSTALL_DIR/embedded/.installed_by_pkg.txt $INSTALL_DIR/embedded/.all-integrations.txt | grep -v '^#' | xargs --no-run-if-empty -I '{}' rm -r $INSTALL_DIR/{}
+
+    rm $INSTALL_DIR/embedded/.all-integrations.txt
+    cd "$PREV_DIR"
+else
+    PIP2_PATH=$INSTALL_DIR/embedded/bin/pip2
+    PIP3_PATH=$INSTALL_DIR/embedded/bin/pip3
+    if [ -x $PIP2_PATH ]; then
+        echo "Uninstalling Python 2 integrations..."
+        $PIP2_PATH freeze | grep ^datadog- | grep -v datadog-checks-base | xargs $PIP2_PATH uninstall -y -q --no-cache-dir || true
+    fi
+    if [ -x $PIP3_PATH ]; then
+        echo "Uninstalling Python 3 integrations..."
+        $PIP3_PATH freeze | grep ^datadog- | grep -v datadog-checks-base | xargs $PIP3_PATH uninstall -y -q --no-cache-dir || true
+    fi
+fi
+
+# Delete all the .pyc/.pyo files in the embedded dir that are part of the old agent's package
+# This MUST be done after using pip or any python, because executing python might generate .pyc files
+if [ -f "$INSTALL_DIR/embedded/.py_compiled_files.txt" ]; then
+    # (commented lines are filtered out)
+    cat $INSTALL_DIR/embedded/.py_compiled_files.txt | grep -v '^#' | xargs rm -f
+fi
+
+exit 0
```

Comment:
All of this is not needed as we're guaranteed to have the installer in compatible agent versions